app/javascript/controllers/refresh_minutes_controller.js

```diff
@@ -1,41 +1,49 @@
 import { Controller } from "@hotwired/stimulus"
+import { format, minutesToMilliseconds } from 'date-fns';
 
 // Connects to data-controller="refresh-minutes"
 export default class extends Controller {
   static targets = ['minutes'];
   static values = {
     active: Boolean,
-    minutes: Number
+    minutes: Number,
+    startTime: String,
+    format: String,
   };
 
   connect() {
-    this.toggleRefresh()
-    window.addEventListener('turbo:render', () => this.toggleRefresh());
+    this.toggleRefresh();
+    window.addEventListener('turbo:morph', (event) => this.toggleRefresh(event));
   }
 
   toggleRefresh() {
-    if (!this.intervalId && this.activeValue) {
-      let totalMinutes = this.minutesValue;
+    this.clearCurrentInterval();
+    if (this.activeValue && !this.intervalId) {
+
       this.intervalId = setInterval(() => {
-        totalMinutes += 1;
-        this.minutesTarget.textContent = this.formatTime(totalMinutes);
-      }, 1000 * 60);
-    } else if (this.intervalId && !this.activeValue) {
-      this._clearInterval()
-    };
+        this.minutesTarget.textContent = this.formatTime();
+      }, 1000);
+    }
   }
 
-  formatTime(mins) {
-    const hours = Math.trunc(mins / 60);
-    const minutes = (mins % 60).toString().padStart(2, "0");
-    return `${hours}:${minutes}`;
+  formatTime() {
+    let diffDate = new Date(1970, 0);
+    const currentDate = new Date(Date.now());
+    const startDate = new Date(this.startTimeValue);
+    const savedTimeMillis = minutesToMilliseconds(this.minutesValue);
+
+    diffDate.setMilliseconds(currentDate - startDate + savedTimeMillis);
+
+    return format(diffDate, this.formatValue);
   }
 
   disconnect() {
-    this._clearInterval()
+    this.clearCurrentInterval();
+    window.removeEventListener('turbo:morph', (event) => this.toggleRefresh(event));
   }
 
-  _clearInterval() {
+  clearCurrentInterval() {
+    if (!this.intervalId) return;
     clearInterval(this.intervalId);
     this.intervalId = undefined;
   }
```

Comment:
I've just noticed a time registration displaying the wrong minutes, details in the video. Sorry for the background noise, there's a puppy making some noise in the background😅

https://www.loom.com/share/7f4c175dba0e46199d0bae514b9ed20f?sid=be818d54-c3d1-41e3-9a4f-0ff761ec4254

As a sidenote, I think it would be great to keep all time calculations on the server-side when moved to the client side. We put ourselves at risk of needing to handle time zones as they could affect our calculations.

Not only that but I've also faced issues where a user with a wrongly set datetime on their side(PC) will affect client-side datetime manipulations/calculations, I'd suggest we keep the calculations we have on the server-side. What do you think?

Reply:
I agree. That sounds more reasonable. Do you know how we do this without sending a whole lot of requests to the server?

Reply:
On render/page load we could pass the stimulus controller the current `minutes` or calculate implement a `current_minutes` for an active time_reg that gets it from `start_time`. This way the stimulus controller only increments a minute from this value, we keep the business logic on the server side.

No need to compute from `started_time` but the `minutes` passed from page load, and we're sure `minutes/60` will always have the right value